src/wallet/wallet.cpp

```diff
@@ -3827,6 +3827,10 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
     if (fFirstRun)
     {
         // ensure this wallet.dat can only be opened by clients supporting HD with chain split and expects no default key
+        if (!gArgs.GetBoolArg("-usehd", true)) {
+            InitError(strprintf(_("Error creating %s: You can't create non-HD wallets with this version."), walletFile));
+            return nullptr;
```

Comment:
Is it me or `walletInstance` is leaked?

Reply:
I believe it is, but that is not a problem since the program just exits here. The same behavior happens earlier too with other `InitError`s.